crates/common/tedge_config/src/tedge_config_cli/models/c8y_software_management.rs

```diff
@@ -0,0 +1,41 @@
+use std::fmt::Display;
+use std::fmt::Formatter;
+use std::str::FromStr;
+
+/// A flag that can be set to auto,
+/// meaning the system will have to detect the appropriate true/false setting
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, Eq, PartialEq, doku::Document)]
+pub enum SoftwareManagementApiFlag {
+    Legacy,
+    Advanced,
+}
+
+#[derive(thiserror::Error, Debug)]
+#[error("Failed to parse flag: {input}. Supported values are: legacy, advanced")]
+pub struct InvalidSoftwareManagementApiFlag {
+    input: String,
+}
+
+impl FromStr for SoftwareManagementApiFlag {
```

Comment:
Since `strum` doesn't provide a solution to change the return error type, if we use `strum` here, we will lose the control of error type, meaning a specific error message for invalid input. So I would keep it as it is unless you know a good solution for that.